packages/web/components/ad-banner/ad-banner-content.spec.tsx

```diff
@@ -0,0 +1,43 @@
+import '@testing-library/jest-dom';
+
+import { render, screen } from "@testing-library/react";
+
+import { AdBannerContent } from "./ad-banner-content";
+import { Ad } from "./ad-banner-types";
+
+const mockAd: Ad = {
+  name: "Mock Ad",
+  start_date: "2023-08-01",
+  end_date: "2023-08-31",
+  header: "Mock Header",
+  subheader: "Mock Subheader",
+  external_url: "https://example.com",
+  icon_image_url: "https://example.com/icon.png",
+  icon_image_alt: "Mock Icon",
+  gradient: "linear-gradient(90deg, #FF0000, #00FF00)",
+  font_color: "#333",
+  arrow_color: "#555",
+  featured: true,
+};
+
+test("renders ad banner content correctly", () => {
+  render(<AdBannerContent {...mockAd} />);
+  const headerElement = screen.getByText(mockAd.header);
+  const subheaderElement = screen.getByText(mockAd.subheader);
+  const imageElement = screen.getByAltText(mockAd.icon_image_alt);
+  const linkElement = screen.getByTestId('ad-banner-link')
+
+  expect(headerElement).toBeInTheDocument();
+  expect(subheaderElement).toBeInTheDocument();
+  expect(imageElement).toBeInTheDocument();
+  expect(linkElement).toHaveAttribute("href", mockAd.external_url);
+});
+
+test("applies correct styles", () => {
+  render(<AdBannerContent {...mockAd} />);
+  const adBannerElement = screen.getByTestId('ad-banner-link')
+  expect(adBannerElement).toHaveStyle({
+    backgroundImage: mockAd.gradient,
+  });
+});
```

Comment:
I'm not in favor of testing implementation details, such as styles, because they don't necessarily represent what the end user will see. If we want to verify elements like gradients, we would require snapshot testing, which provides a more accurate representation of the user interface. If you'd like to read more on this, check Kent C. Dodds, creator of React Testing Library, [article](https://kentcdodds.com/blog/testing-implementation-details) about it